src/main/java/seedu/address/model/student/Student.java

```diff
@@ -27,40 +27,68 @@ public class Student {
     // Data fields
     private final Memo memo;
 
+    private Class studentClass = null;
+
     // The current lesson's attendance (Present/Absent)
     private CurrentLessonAttendance currentLessonAttendance;
     // Cumulative number of lessons attended
     private LessonsAttended lessonsAttended;
 
     /**
-     * If only name is provided.
+     * Constructs a Student with Name and Index that represents one-based class index.
+     *
+     * @param name Name to represent Student.
      */
     public Student(Name name) {
-        requireNonNull(name);
+        requireAllNonNull(name);
```

Comment:
Perhaps we could just stick with requireNonNull here since only one argument is passed